graphql/client.go

```diff
@@ -38,10 +38,38 @@ type Client interface {
 		req *Request,
 		resp *Response,
 	) error
+
+	// DialWebSocket must open a webSocket connection and subscribe to an endpoint
+	// of the client's GraphQL API.
+	//
+	// ctx is the context that should be used to make this request.  If context
+	// is disabled in the genqlient settings, this will be set to
+	// context.Background().
+	//
+	// req contains the data to be sent to the GraphQL server. Will be marshalled
+	// into JSON bytes.
+	//
+	// respChan is a channel used to send the data that arrives via the
+	// webSocket connection.
+	//
+	// errChan is a channel on which are sent the errors of webSocket
+	// communication.
+	//
+	// err is any error that occurs when setting up the webSocket connection.
+	DialWebSocket(
+		ctx context.Context,
+		req *Request,
+		respChan chan json.RawMessage,
+	) (errChan chan error, err error)
```

Comment:
Why is `respChan` a parameter but `errChan` a return? Is this a standard thing?

Also, I definitely regretted not initially making `Response` a struct -- thoughts on changing this to `resp *WebsocketResponse` which would have `respChan` (and maybe `errChan`) as fields so we can add more later?

Reply:
I did it this way because the `respChan`'s type depends on the type of the returned object, while the `errChan` always sends an error, so it could be added to the `webSocketClient` struct.
But if it is disturbing, I could bring back the `errChan` as a parameter, just like for the `respChan`.